assets/javascripts/discourse/widgets/discourse-reactions-reaction-button.js.es6

```diff
@@ -75,6 +75,7 @@ export default createWidget("discourse-reactions-reaction-button", {
       title = "discourse_reactions.main_reaction.cant_remove";
     }
 
+    // !likeAction.hasOwnProperty("can_undo")  it specifies user have not liked the post
```

Comment:
no that's not the comment I was expected :) why do you do:

```
!likeAction.hasOwnProperty("can_undo")
```

and not

```
!likeAction.can_undo
```

?

This is what you should explain

Reply:
how about this?
hasOwnProperty("can_undo") can_undo property is included only if the post is liked by user. So, it is used here to check whether users have not liked the post or not irrespective of the can_undo value which can be true or false. Liked posts can't be assigned the title = "discourse_reactions.picker.remove_reaction" OR "discourse_reactions.picker.cant_remove_reaction".

Reply:
OR this
used `!likeAction.hasOwnProperty("can_undo")` rather than `!likeAction.can_undo` to check ther can_undo property is present or not irrsepective of its value